### PR TITLE
Add embedding op to jagged NT

### PR DESCRIPTION
**Pull request by ani300**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112288

### CODE REVIEW

File: test/test_nestedtensor.py [hunk 1]
```diff
@@ -3044,6 +3044,18 @@ def test_unbind(self, device):
             for i, t in enumerate(out):
                 self.assertEqual(t, tensor_list[i])
 
+    def test_embedding(self, device):
```
Comment (cpuhrsch) [inline test/test_nestedtensor.py:3047]: There is already a test for this https://github.com/pytorch/pytorch/blob/main/test/test_nestedtensor.py#L795 but for the other layout. Can we unify these two?

**Review by cpuhrsch — APPROVED** [review]
Seems reasonable to me.

File: test/test_nestedtensor.py (hunk 1 above)
Comment (ani300) [reply test/test_nestedtensor.py:3047]: I wrote the original test too... lol. But any suggestions on how to better unify the tests? I guess one option is adding an extra parameter to the tests for the layout in a new class or something, but I saw some mention of OpInfo in the comments of the file (which I don't know anything about)

Comment (cpuhrsch) [reply test/test_nestedtensor.py:3047]: Haha, that's right! Full circle.

Eventually we can migrate some of these tests to opinfos (we can't reimplement all tests for NT as well).

I think adding an extra parameter for layout is a good start at least in context of this PR.

Comment (cpuhrsch) [reply test/test_nestedtensor.py:3047]: It's ok to do this in a follow up PR as well. We can merge this one for now as is.

**Pull request closed without merging**